src/poseidon/traits.rs

```diff
@@ -0,0 +1,416 @@
+use crate::poseidon::grain_lfsr::PoseidonGrainLFSR;
+use crate::poseidon::PoseidonParameters;
+use ark_ff::{fields::models::*, FpParameters, PrimeField};
+use ark_std::{vec, vec::Vec};
+
+/// A trait for default Poseidon parameters associated with a prime field
+pub trait PoseidonDefaultParameters: FpParameters {
+    /// An array of the parameters optimized for constraints
+    /// (rate, alpha, full_rounds, partial_rounds, skip_matrices)
+    /// for rate = 2, 3, 4, 5, 6, 7, 8
+    ///
+    /// Here, `skip_matrices` denote how many matrices to skip before
+    /// finding one that satisfy all the requirements.
+    const PARAMS_OPT_FOR_CONSTRAINTS: [[usize; 5]; 7];
+
+    /// An array of the parameters optimized for weights
+    /// (rate, alpha, full_rounds, partial_rounds, skip_matrices)
+    /// for rate = 2, 3, 4, 5, 6, 7, 8
+    const PARAMS_OPT_FOR_WEIGHTS: [[usize; 5]; 7];
```

Comment:
Instead of using `[usize; 5]`, let's make a struct and use that?

Reply:
Let me have a try.